TestApp/Model/NoteModel.swift

```
@@ -5,7 +5,21 @@
 //
 
 import Foundation
+import CoreData
 
-class NoteModel {
+extension NoteModel {
+
+    override public func awakeFromInsert() {
+        super.awakeFromInsert()
+        self.date = Date() as NSDate
+    }
+    
+    static func object() -> NoteModel? {
```

You probably don't want a nil data object back, this should throw an error.